backends/gstreamer/src/media_stream_source.rs

```diff
@@ -0,0 +1,250 @@
+use glib;
+use glib::subclass;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use gst;
+use gst::prelude::*;
+use gst::subclass::prelude::*;
+use media_stream::{GStreamerMediaStream, StreamType};
+use url::Url;
+
+// Implementation sub-module of the GObject
+mod imp {
+    use super::*;
+
+    pub struct ServoMediaStreamSrc {
+        cat: gst::DebugCategory,
+        audio_proxysrc: gst::Element,
+        video_proxysrc: gst::Element,
+        audio_srcpad: gst::GhostPad,
+        video_srcpad: gst::GhostPad,
+    }
+
+    impl ServoMediaStreamSrc {
+        pub fn set_audio_stream(&self, stream: &GStreamerMediaStream) {
+            debug_assert!(stream.type_() == StreamType::Audio);
+            gst_log!(self.cat, "Setting audio stream");
+            // XXX Do proxysrc stuff.
```

Comment:
What _should_ work is simply using [`attach_to_pipeline`](https://github.com/servo/media/blob/62d748a799e1df240b661d37b758ced9830ef999/backends/gstreamer/src/media_stream.rs#L94-L103) and then linking the last element to your player element [like so](https://github.com/servo/media/blob/62d748a799e1df240b661d37b758ced9830ef999/backends/gstreamer/src/media_stream.rs#L223)